### ISSUE TITLE
improve clij3 architecture

### ISSUE THREAD

**Issue by StRigaud** [clEsperanto/clij3#8]
For now all the code is in the `CLIJ3.java` file. 

Could it be possible to split the core code and the auto-generated code into 2 files? To facilitate maintenance and development

CC @haesleinhuepf

**Comment by haesleinhuepf** [comment]
I would also say, we should move the CLIJ3 class into a clij3 package. Does anything speak agains this?

**Comment by StRigaud** [comment]
i don't know, maybe @tinevez or @carlosuc3m  can give us a hint on this

**Comment by tinevez** [comment]
I would stick to the 'split the class if it is >1000 lines'.
We want to promote maintainability, so splitting core vs autogenerated makes sense as well.
Does it make sense not to?

**Comment by haesleinhuepf** [comment]
Yes splitting makes sense! The auto-generated part was also in a separate file in CLIJ2.

**Comment by carlosuc3m** [comment]
yess, better to split it!!
